teps/0040-ignore-step-errors.md

```diff
@@ -145,13 +156,306 @@ This proposal is limited to a step within a task and does not address `pipelineT
 * A [platform team](https://github.com/tektoncd/community/blob/main/user-profiles.md#1-pipeline-and-task-authors)
   wants to share a `Task` with their team which runs the following steps in a sequence:
   * Run unit tests (which may fail)
-  * Apply a transformation to the test results (e.g. converts them to a certain format such as junit)
+  * Apply a transformation to the test results (e.g., converts them to a certain format such as junit)
   * Upload the results to a central location used by all the teams
 
+## Proposal
+
+Introduce a new section `exit` as part of the
+[steps](https://github.com/tektoncd/pipeline/docs/tasks.md#defining-steps) definition along with an `image` and `script`.
+
+```
+- name: failing-step
+  image: alpine
+  exit:
```

Comment:
Last nit: any reason to not have `onError` directly under the step spec ?

Reply:
Thanks @vdemeester thats definitely a great idea and a little less verbose:

```yaml
steps:
  - name: my-step
     onError:  [ continue | fail ]
```

Compared to:

```yaml
steps:
  - name: my-step
    exit:
       onError:  [ continue | fail ]
```

The goal behind creating a separate section for `exit` was to extend it to add any `exit` related specifications in future, for example:

```yaml
steps:
  - name: my-step
    exit:
       onError:  [ continue | fail ]
       exitCode: 1
```

I am open to either of these specifications, @bobcatfish thoughts?

Reply:
i DO really like the simplicity of ` onError:  [ continue | fail ]` - like you said @pritidesai it feels like it's a matter of whether we feel like we might add more configuration around this later or not, which we'd want grouped under some section like `exit`

i suppose on the positive side, if we went with just ` onError:  [ continue | fail ]` and wanted to add more later, it could be an additive change e.g. we support both specifying ` onError:  [ continue | fail ]` and specifying it in an `exit` section

i think it comes down to how likely we feel it is that we'll want to add more - if very likely, keep the `exit` section, if not, keep it simple and just do ` onError:  [ continue | fail ]`

i dont have a strong opinion either way!

Reply:
Awesome, lets keep it simple for now 😄  I have updated the proposal to introduce `onError` at the step specification level. Thank you both ❤️